r2-core/src/main/java/com/linkedin/r2/message/timing/TimingKey.java

```diff
@@ -79,7 +84,7 @@ private static TimingKey registerNewKey(TimingKey timingKey)
   {
     if (_pool.putIfAbsent(timingKey.getName(), timingKey) != null)
     {
-      throw new IllegalStateException("Timing key " + timingKey.getName() + " has already been registered!");
+      LOG.warn("Timing key " + timingKey.getName() + " has already been registered!");
```

Comment:
Given that now we have unique key, this exception should be kept, right?

Reply:
github doesn't let us reply to the top level comment.

"I'm curious why we're getting duplicated keys. Was it because this filter is instantiated twice? That sounds an error to me."
If you have multiple clients it's possible a collision happens. This is more likely in offline jobs where they spawn many clients, but it can also happen for online apps. 

I'd still rather log a warning than fail deployment for future potential edge cases like this one.

Reply:
What edge case we're talking about? We have logic to make the key unique. If it is not, it's a bug. We should fix it instead of let it go with a warning. What's wrong if we fail the deployment like this?

Reply:
Imo it's not a severe enough error to justify failing the deployment. The app can still operate completely ok - just the timings may be inaccurate.

Reply:
OK